plugins/woocommerce/client/blocks/assets/js/blocks/featured-items/with-featured-item.tsx

```diff
@@ -114,17 +116,75 @@ export const withFeaturedItem =
 			product,
 			setAttributes,
 		} = props;
-		const { mediaId, mediaSrc } = attributes;
+		const { mediaId, mediaSrc, isRepeated, imageFit, backgroundColor } =
+			attributes;
 		const item = category || product;
 		const [ backgroundImageSize, setBackgroundImageSize ] = useState( {} );
 
-		const { backgroundImageSrc } = useBackgroundImage( {
+		const {
+			backgroundImageSrc,
+			setFeaturedProductParentDivDimensions,
+			backgroundColorVisibility,
+		} = useBackgroundImage( {
 			item,
 			mediaId,
 			mediaSrc,
 			blockName: name,
+			blockAttributes: { isRepeated, imageFit },
 		} );
 
+		const blockName =
+			name === 'woocommerce/featured-category'
+				? 'Featured Category'
+				: 'Featured Product';
+
+		useEffect( () => {
+			// Observes the resizable block's dimension changes.
+			const observer = new ResizeObserver( ( entries ) => {
+				setFeaturedProductParentDivDimensions( {
+					height: entries[ 0 ].contentRect.height,
+					width: entries[ 0 ].contentRect.width,
+				} );
+			} );
+
+			if ( isLoading === false ) {
+				const element = document
+					?.getElementsByTagName( 'iframe' )[ 0 ]
+					?.contentWindow?.document.querySelector(
+						'.featured-product-parent'
+					) as HTMLElement | null;
```

Comment:
Also, if possible, we should avoid using the query selector but, instead, we should use `ref` to control DOM elements: https://react.dev/learn/manipulating-the-dom-with-refs